BnbnavNetClient/Services/NetworkOperations/RoadUpdateOperation.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Avalonia.Media;
+using BnbnavNetClient.Models;
+using BnbnavNetClient.Views;
+
+namespace BnbnavNetClient.Services.NetworkOperations;
+
+public class RoadUpdateOperation : NetworkOperation
+{
+    private readonly MapEditorService _editorService;
+    private readonly Road _road;
+    private readonly string _newName;
+    private readonly RoadType _newRoadType;
+    private readonly List<Edge> _affectedEdges = new();
+
+    public RoadUpdateOperation(MapEditorService editorService, Road road, string newName, RoadType newRoadType)
+    {
+        _editorService = editorService;
+        _road = road;
+        _newName = newName;
+        _newRoadType = newRoadType;
+        
+        _affectedEdges.AddRange(editorService.MapService!.Edges.Values.Where(x => x.Road == road));
+    }
+    
+    public override async Task PerformOperation()
+    {
+        ItemsNotToRender.AddRange(_affectedEdges);
+        
+        try
+        {
+            (await _editorService.MapService!.Submit($"/roads/{_road.Id}", new
+            {
+                name = _newName,
+                type = _newRoadType.ServerName()
+            })).AssertSuccess();
+        }
+        catch (HttpRequestException)
+        {
```

Comment:
i'm guessing handling for this is planned?

Reply:
There's nothing to handle right now but if we want to show an error message to the user that's where we'll raise it